Introduce MessageProfiles to v1.6 load tests [CCIP-5143]

asoliman92 opened the pull request:
Introduce MessageProfiles to load tests

asoliman92 committed:
Initial commit

Add MessageDetails param to enable different message types

asoliman92 committed:
Get message in destination_gun based on MessageDetails

Also set the gasLimit based on the MessageDetails

asoliman92 committed:
Increase FeeQuoter MaxDataBytes

asoliman92 committed:
Add documentation

asoliman92 committed:
Add initial configuration

asoliman92 committed:
Merge branch 'develop' into load-msg-details

asoliman92 committed:
linting

asoliman92 committed:
Increase maxdatabytes

asoliman92 committed:
Add realistic config

asoliman92 committed:
Don't override gasLimit for token message types

asoliman92 committed:
Merge branch 'develop' into load-msg-details

makramkd committed:
go.mod: bump cl-ccip (#16561)

Has a fix for https://smartcontract-it.atlassian.net/browse/CCIP-5312

asoliman92 committed:
linting

asoliman92 committed:
More proper values for FeeQuoter dest chain config

eutopian committed:
[CAPPL-576] use workflow key instead of evm config key (#16517)

* use workflow key instead of evm config key as encryption pub key

* update jd version in e2e tests

0xAustinWang commented on integration-tests/load/ccip/destination_gun.go:
```diff
@@ -76,13 +76,9 @@ func NewDestinationGun(
-		sum += weight
+	for _, md := range *m.testConfig.MessageDetails {
+		sum += *md.Ratio
 	}
```
let's do this just in testconfig/ccip/load.go

asoliman92 replied:
I agree, I saw it being done here as well so left it as is, in this case I'll remove the Validate function from destGun as it won't be checking anything else

0xAustinWang commented on integration-tests/load/ccip/destination_gun.go:
```diff
@@ -184,54 +181,61 @@ func (m *DestinationGun) MustSourceChain() (uint64, error) {
 // GetMessage will return the message to be sent while considering expected load of different messages
-func (m *DestinationGun) GetMessage(src uint64) (router.ClientEVM2AnyMessage, error) {
+// returns the message, gas limit
+func (m *DestinationGun) GetMessage(src uint64) (router.ClientEVM2AnyMessage, int64, error) {
```
can we add an early return if there's only a single message type?

asoliman92 replied:
I think it's a non necessary optimization IMO. we're talking a couple of milliseconds at max.

0xAustinWang commented on integration-tests/load/ccip/destination_gun.go:
```diff
@@ -109,10 +105,14 @@ func (m *DestinationGun) Call(_ *wasp.Generator) *wasp.Response {
+	// Set the gas limit for this tx
+	if gasLimit != 0 {
+		acc.GasLimit = uint64(gasLimit)
+	}
```
do we need to reset this in case a message profile wants to use default gas limit?

asoliman92 replied:
If it's zero then it's gonna use the default

asoliman92 commented on deployment/ccip/changeset/v1_6/cs_chain_contracts.go:
```diff
@@ -1721,20 +1721,20 @@ func DefaultFeeQuoterDestChainConfig(configEnabled bool, destChainSelector ...ui
 		IsEnabled:                         configEnabled,
 		MaxNumberOfTokensPerMsg:           10,
-		MaxDataBytes:                      256,
+		MaxDataBytes:                      30_000,
```
copied from 1.5 mainnet values.

mateusz-sekara replied:
Do we need someone from onchain team to double check these values?

mateusz-sekara replied:
It's very concerning that this is the first time we noticed invalid values inthe  changeset which were used for deployments on staging, no?

asoliman92 replied:
Yeah. I already had the green light from @RensR  about these numbers. We didn't catch in staging because we don't send bigger messages

asoliman92 committed:
Remove duplicate validation

justinkaseman committed:
Change default Workflow Execution Rate Limits to PerSender 200 (#16549)

* Change default Workflow Execution Rate Limits to PerSender 200

* Regen docs

krehermann committed:
feat(BCF-3477): support importing keys via TOML secrets (#16505)

* wip: support single eth key secret

* import p2p key

* support multiple eth keys

* fix eth keys plumbing and add test

* rename vars

* working locally

* cleanup

* address feedback

* linter

bukata-sa committed:
CCIP-4796 ccip analyze mcms proposal (#16384)

* CCIP-4796 ccip analyze mcms proposal

* mcms analyze

* integrate into deployer group

* review fixes

* review fixes

samsondav committed:
ReportCodec verification of channel definitions; bump go to 1.24 (#16541)

* ReportCodec verification of channel definitions

* Bump go to 1.24

* remove obsolete cluster profile

* Fix mockery

* Fix tests

---------

Co-authored-by: skudasov <f4hrenh9it@gmail.com>

tt-cll committed:
add update dest configs (#16532)

* add timelock to remote chain

* lint

* re-add transfer

* move to mcms helper

* fix buffer bug

* mcms validation

* set authority

* refactor

* try to unify ci and local

* lint

* add update dest configs

* add disable

* cleanup

* error message

* v2

* fix test

* add mcms to billing

* refactor into helper

* lint

* test without mcms

pavel-raykov committed:
Fix abigen compilation on Mac. (#16564)

* Minor

* Minor

skudasov committed:
bump CTF to Go 1.24 commit (#16566)

* bump CTF to Go 1.24 commit

* use old Seth version

* re-trigger

* update again

eutopian committed:
move workflow key nil check to after existing node check to support old nodes (#16569)

ecPablo committed:
feat: update mcms lib with new timelock converter and update BuildProposalFromBatches() for solana chain metadata (#16528)

* feat: update mcms lib with new timelock converter and update BuildProposalFromBatchesV2() to accept env for extracting solana metadata.

* feat: update go mod to new mcms and ccip-solana version

* feat: update go mod to new mcms and ccip-solana version

* fix: go mod

* fix: go mod

* fix: go mod core scripts

* chore: add changeset

* fix: update tx field name to RawData

* fix: more cleanup

- update tests
- go mod tidy
- delete comments
- other bug fixes

* fix: break dep cycle

for state package, stop relying on proposalutils package so the `buildProposalMetadataV2` can call `MaybeLoadMCMSWithTimelockChainStateSolana`

* fix: merge conflicts develop

* fix: go sum

* fix: update lib

* fix: upgrade ccip solana

* fix: downgrade ccip solana for rest of core packages and chainlink/deployment

* fix: go mod core scripts add replace

* fix: downgrade solana from mcms lib

* fix: merge conflicts develop

* fix: merge conflicts develop

* fix: go mod deps

* fix: params in solana utils

---------

Co-authored-by: Graham Goh <graham.goh@smartcontract.com>

winder committed:
Add README that tells you how to run a CCIP test. (#16233)

mateusz-sekara commented on integration-tests/testconfig/ccip/ccip.toml:
```diff
@@ -242,14 +242,23 @@ addresses_to_fund = [
+# Messages that include data MUST have a `DataLengthBytes` field
+MessageDetails = [
+     {Ratio = 100, MsgType = "Token"},
+#    {Ratio = 3, MsgType = "Token"},
```
This commented profile should be the default one, right?

asoliman92 replied:
Yes. Going to uncomment 👍

mateusz-sekara commented on integration-tests/testconfig/ccip/ccip.toml:
```diff
@@ -242,14 +242,23 @@ addresses_to_fund = [
 [Load.CCIP.Load]
-# MessageTypeWeights corresponds with [data only, token only, message with token]
-MessageTypeWeights = [10,45,45]
+# We have 3 types of messages: Token, DataWithToken, Data
```
It would be better to speak the same language and use the product glossary (this is how the features are named). Otherwise, it leads to confusion and drift between product and implementation. Thus, I suggest renaming to:
* Token Transfer (Token)
* Programmable Token Transfer (DataWithToken)
* Messaging (Data)

mateusz-sekara commented on integration-tests/load/ccip/destination_gun.go:
```diff
@@ -184,54 +165,61 @@ func (m *DestinationGun) MustSourceChain() (uint64, error) {
+	m.l.Infow("Selected message type", "msgType", *selectedMsgDetails.MsgType)
+
+	message := router.ClientEVM2AnyMessage{
+		Receiver:  rcv,
```
nit, but Data and DataWithToken must be send to contract (I guess it's rcv), but TokenTransfer could be send to any EOA

asoliman92 replied:
It doesn't matter for the sake of this test, right? What are you suggesting? Sending it to the dest.DeployerKey for example?

mateusz-sekara replied:
You can even send that `evmutils.RandomAddress()` or `DeployerKey`, I only suggest to not send to contract but to EOA

mateusz-sekara commented on integration-tests/load/ccip/destination_gun.go:
```diff
@@ -184,54 +165,61 @@ func (m *DestinationGun) MustSourceChain() (uint64, error) {
+		message.TokenAmounts = []router.ClientEVMTokenAmount{
+			{
+				Token:  m.state.Chains[src].LinkToken.Address(),
+				Amount: big.NewInt(1),
```
Will the sender run out of tokens at some stage? I guess the question is, what's the initial number of tokens we mint for senders

asoliman92 replied:
We mint smth with e18, so shouldn't run out of tokens with us sending 1 by 1 like that.

mateusz-sekara commented on integration-tests/testconfig/ccip/ccip.toml:
```diff
@@ -242,14 +242,23 @@ addresses_to_fund = [
 # Directory where we receive environment configuration from crib
 CribEnvDirectory = "../../../../crib/deployments/ccip-v2/.tmp"
-TimeoutDuration = "30m"
-GasLimit = 700_000
```
👏

mateusz-sekara commented on integration-tests/load/ccip/destination_gun.go:
```diff
@@ -184,54 +165,61 @@ func (m *DestinationGun) MustSourceChain() (uint64, error) {
+	accumulatedRatio := 0
+	var selectedMsgDetails *ccip.MsgDetails
+
+	for _, msg := range *m.testConfig.MessageDetails {
```
Is there any way to test it out in an automated way (e.g. unit test)? Or maybe exposing metrics (or logs) with a number of different messages sent? I'm afraid it will be very easy to accidentally introduce some regression without noticing

asoliman92 replied:
I'm exposing a log [here](https://github.com/smartcontractkit/chainlink/pull/16556/files#diff-9e74374ec60e2e8c6a106dea07b02c7b10939d30aa76ea8d5e30577a2385690eR193). Metrics will be better, gonna check adding a metric as well.

pavel-raykov committed:
Remove bash wrappers and invoke solidity compilation directly. (#16568)

* Minor

* Minor

asoliman92 committed:
debug logs (#16581)

asoliman92 committed:
Use product names

makramkd committed:
.github: fix integration in memory tests workflow (#16586)

* .github: fix integration in memory tests workflow

* fix

mateusz-sekara commented on integration-tests/testconfig/ccip/ccip.toml:
```diff
@@ -242,14 +242,23 @@ addresses_to_fund = [
+    {Ratio = 3, MsgType = "Token"},
+    {Ratio = 82, MsgType = "DataWithToken", DataLengthBytes = 5000, DestGasLimit=650_000},
+    {Ratio = 10,  MsgType = "Data", DataLengthBytes = 10_000, DestGasLimit=800_000},
+    {Ratio = 5,  MsgType = "Data", DataLengthBytes = 10_000, DestGasLimit=2_500_000},
```
Please fix ;)

asoliman92 committed:
Use product names in config

winder committed:
Rename commitTransmitter to ccipTransmitter (#16235)

asoliman92 committed:
Use EOA with token transfers only
